### ISSUE TITLE
Validating URL from another server

### ISSUE THREAD

**Issue by timgavin** [spatie/url-signer#7]
I'm using Spatie Media Library for my images and am generating private and public URLs using Spatie URL Signer (Laravel package). This is working perfectly as long as everything is on the same server.

Recently I've started thinking about integrating a load balancer, which means I'll have to host my images on another server. Instead of using Amazon I've decided to go with DigitalOcean's block storage.

Now my problem: how do I validate a URL on server B that was generated on Server A?

Server B is my media server. To keep things light I'm working with the non-Laravel version of URL Signer on server B.

I've tried using the same keys on both servers (something really simple for testing), but that's not working.

```
SERVER A (Laravel)
config/laravel-url-signer.php
    'signatureKey' => '123123123',

SERVER B (PHP)
use Spatie\UrlSigner\MD5UrlSigner;
$urlSigner = new MD5UrlSigner('123123123');
```

No matter what I do the validation just keeps returning false. Is this possible?

**Comment by freekmurze** [comment]
This [line of code](https://github.com/spatie/url-signer/blob/master/src/MD5UrlSigner.php#L19) is used to create the signature. It's also used to verify the signature.

Could you verify the parameters being passed in and the generated output are the same when generating the signature on server A and using the link on server B. 

Just put some `dump` statement in the code of the package in the `vendor` directory on both servers.

**Comment by timgavin** [comment]
OK, so I did find one discrepancy. I had two URL Generator packages installed; the older one and the newer one. Since the newer version was renamed (removing 'laravel') I didn't realize I still had the old one installed, and that's the one where I was modifying the `signaturekey`, so obviously it wasn't working with the new parameter `123123123`.

I located the correct copy (`vendor/spatie/url-signer/src/MD5UrlSigner.php`) and modified the `signatureKey` and did

```
protected function createSignature($url, $expiration)
    {
        $url = (string) $url;
        
        die($this->signatureKey);

        return md5("{$url}::{$expiration}::{$this->signatureKey}");
    }
```

Which outputs `123123123`, so it appears to be working correctly. However, it's still returning `false` on server B.

-----

When generating a new link on Server B, the signature is different from Server A, but then again, the signature is different every time I generate a new link. I'm assuming this is expected behavior?

-----

What I'm doing on Server A is copying the image address (right-click the image icon in the browser and select Copy Image Address) and then pasting that into the browser to verify on Server B, which should show the image.
